log.go

```diff
@@ -0,0 +1,38 @@
+package artifact
+
+import (
+	"io"
+	"log"
+	"os"
+)
+
+// By default, we're creating a logger which is used to print to standard
+// output.  We are doing to ensure that by default, consumers of this library
+// get useful logs for debugging.  We do, however, want to allow the users to
+// change where logging should go, so we'll expose methods to change the
+// behaviour of logging.
+var logger = log.New(os.Stdout, "artifact:", log.LUTC)
+
+// SetLogOutput will change the prefix used by logs in this package This is a
+// simple convenience method to wrap this package's Logger instance's method.
+// See: https://golang.org/pkg/log/#Logger.SetOutput
+// If you wish to unconditionally supress all logs from this library, you can
+// do the following:
+//  SetLogOutput(ioutil.Discard)
+func SetLogOutput(w io.Writer) {
+	logger.SetOutput(w)
+}
+
+// SetLogPrefix will change the prefix used by logs in this package This is a
+// simple convenience method to wrap this package's Logger instance's method.
+// See: https://golang.org/pkg/log/#Logger.SetPrefix
+func SetLogPrefix(p string) {
+	logger.SetPrefix(p)
+}
+
+// SetLogFlags will change the flags which are used by logs in this package
+// This is a simple convenience method to wrap this package's Logger instance's
+// method.  See: https://golang.org/pkg/log/#Logger.SetFlags
+func SetLogFlags(f int) {
+	logger.SetFlags(f)
+}
```

Comment:
It is a neat trick to just expose only wrapper functions that alter the logging options rather than exporting or exposing the entire underlying logger. I guess this is to prevent that someone logs to the internal logger? Would it be useful to have a function where you can pass in a custom `log.Logger` in case you have set one up for your entire application, and want your application dependencies to write to the same log with the same flags and prefix? This may save a few lines for users of the library that already have a custom log.Logger that they are using?